packages/vuelidate/src/core.js

```diff
@@ -279,25 +268,29 @@ function createValidationResults (rules, state, key, parentKey) {
  * @return {{}}
  */
 function collectNestedValidationResults (validations, state, key) {
-  const nestedValidationKeys = Object.keys(validations)
-
-  // if we have no state, return empty object
-  if (!nestedValidationKeys.length) return {}
-
-  return nestedValidationKeys.reduce((results, nestedKey) => {
-    // if we have a key, use the nested state
-    // else use top level state
-    const nestedState = key ? state[key] : state
-
-    // build validation results for nested state
-    results[nestedKey] = setValidations({
-      validations: validations[nestedKey],
-      state: nestedState,
-      key: nestedKey,
-      parentKey: key
-    })
-    return results
-  }, {})
+  const collected = ref({})
+
+  watch(validations, (changedValidations) => {
+    const nestedValidationKeys = Object.keys(changedValidations)
+    // if we have no state, return empty object
+    if (!nestedValidationKeys.length) return
+
+    collected.value = nestedValidationKeys.reduce((results, nestedKey) => {
+      // if we have a key, use the nested state
+      // else use top level state
+      const nestedState = key ? state[key] : state
```

Comment:
More of a comment to the initial implementation of this, but I’m wondering – shouldn’t we use `nestedKey` here? Given that state and validation rules should map 1:1, we should be able to use the same keys, right? Unless one of those two is behind when it comes the "depth" we are currently. Wdyt?

Reply:
Nested key is the name of the validator, not the property it validates, right?